harvester_e2e_tests/integration/test_rancher_integration.py

```diff
@@ -0,0 +1,409 @@
+# Copyright (c) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 3 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com
+
+from time import sleep
+from datetime import datetime, timedelta
+
+import pytest
+
+
+pytest_plugins = [
+    'harvester_e2e_tests.fixtures.api_client',
+    'harvester_e2e_tests.fixtures.rancher_api_client',
+]
+
+
+@pytest.fixture(scope="session")
+def rancher_wait_timeout(request):
+    return request.config.getoption("--rancher-cluster-wait-timeout")
+
+
+@pytest.fixture(scope='class')
+def harvester_cluster_name(unique_name):
+    return f"{unique_name}-harv"
+
+
+@pytest.fixture(scope='class')
+def rke1_cluster_name(unique_name):
+    return f"{unique_name}-rke1"
+
+
+@pytest.fixture(scope='class')
+def rke2_cluster_name(unique_name):
+    return f"{unique_name}-rke2"
+
+
+@pytest.fixture(scope='session')
+def vlan_network(request, api_client):
+    vlan_nic = request.config.getoption('--vlan-nic')
+    vlan_id = request.config.getoption('--vlan-id')
+    # don't create network if VLAN is not correctly specified
+    if vlan_id == -1:
+        return
```

Comment:
I'm thinking there might have a hidden issue if the vlan will not be created and still pass the fixture setup 🤔

Reply:
Updated it. Thank you.